src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,11 +584,341 @@ export default class TransformationPanel extends Panel {
     this.update();
   }
 
+  _splitSelection(layerGlyphController, selection) {
+    let { point: pointIndices, component: componentIndices } =
+      parseSelection(selection);
+    pointIndices = pointIndices || [];
+
+    const points = [];
+    const contours = [];
+    const components = componentIndices || [];
+
+    if (!pointIndices.length) {
+      return { points, contours, components };
+    }
+
+    const path = layerGlyphController.instance.path;
+    const pathSelection = filterPathByPointIndices(
+      layerGlyphController.instance.path,
+      pointIndices
+    );
+    const selectionByContour = getSelectionByContour(path, pointIndices);
+
+    let contourIndex = 0;
+    for (const pointIndex of pointIndices) {
+      while (path.contourInfo[contourIndex].endPoint < pointIndex) {
+        contourIndex++;
+      }
+
+      let pathSelectionContourIndex;
+      for (const [j, [cIndex, contourSelection]] of enumerate(selectionByContour)) {
+        if (contourIndex === cIndex) {
+          pathSelectionContourIndex = j;
+          break;
+        }
+      }
+
+      if (pathSelection.contourInfo[pathSelectionContourIndex].isClosed) {
+        const contourStartIndex = !contourIndex
+          ? 0
+          : layerGlyphController.instance.path.contourInfo[contourIndex - 1].endPoint +
+            1;
+        const contourEndIndex = path.contourInfo[contourIndex].endPoint + 1;
+
+        const contourPoints = Array.from(range(contourStartIndex, contourEndIndex));
+
+        if (contourStartIndex === pointIndex) {
+          // only add list of contours
+          // if the point is the start of the contour
+          contours.push(contourPoints);
+        }
+      } else {
+        points.push(pointIndex);
+      }
+    }
+
+    return { points, contours, components };
+  }
+
+  _collectMovableObjects(moveDescriptor, controller) {
+    const { points, contours, components } = this._splitSelection(
+      controller,
+      this.sceneController.selection
+    );
+
+    const movableObjects = [];
+    for (const pointIndex of points) {
+      const individualSelection = [`point/${pointIndex}`];
+      movableObjects.push(new MovablePoint(pointIndex, individualSelection));
+    }
+    for (const [contourIndex, pointIndices] of enumerate(contours)) {
+      const individualSelection = pointIndices.map(
+        (pointIndex) => `point/${pointIndex}`
+      );
+      movableObjects.push(new MovableContour(contourIndex, individualSelection));
+    }
+    for (const componentIndex of components) {
+      const individualSelection = [`component/${componentIndex}`];
+      movableObjects.push(new MovableComponent(componentIndex, individualSelection));
+    }
+
+    let movableObjectsSorted = movableObjects.sort(
```

Comment:
Btw this can be a `const`.

Reply:
Btw 2 I don't quite understand why you unconditionally sort horizontally: it is not necessary _unless_ we distribute horizontally.

Suggestion for refactor: instead of a string property, the moveDescriptor could provide an optional sort function. This would work like this in this section of the code:
```js
    if (moveDescriptor.sortObjects) {
      movableObjects.sort((a, b) => moveDescriptor.sortObjects(a, b, controller));
    }
```
Btw 3: `array.sort()` sorts in-place, and returns itself, so it is redundant to use a different variable for the sorted objects list. We can ignore the return value.